### PR TITLE
[flutter_tools] refactor iOS launch code into InstallService class

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

The device classes do too much and have too many dependencies. This PR explores an example of a gradual migration away from this pattern, where the installation logic of the IOSDevice is moved to an installation service.

In the future, the service need not be a property of the iOS device, and instead could be looked up separately based on the device type.

### CODE REVIEW

File: packages/flutter_tools/lib/src/ios/devices.dart [hunk 1]
```diff
@@ -159,12 +154,197 @@ class IOSDevice extends Device {
...
+  Future<bool> isLatestBuildInstalled(IOSApp app) => _installationService.isLatestBuildInstalled(this, app);
+
+  @override
+  Future<bool> installApp(IOSApp app) => _installationService.installApp(this, app);
+
+  @override
+  Future<bool> stopApp(IOSApp app) => _installationService.stopApp(this, app);
+
+  @override
+  Future<bool> uninstallApp(IOSApp app) => _installationService.uninstallApp(this, app);
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/ios/devices.dart:169]: I guess it makes less sense refactor this way if you think of device as being this stateful handle to a physical device. But yeah, the above is one example of what we could do with this change.

FlutterDevice is not long for this world anyway, its kind of a bad abstraction for multiple devices and we can't even simultaneously run on web/fuchsia

[the remark this answers is not in the document]
Comment (jmagman) [reply packages/flutter_tools/lib/src/ios/devices.dart:169]: > FlutterDevice is not long for this world anyway

I'll believe it when I see it. 😄

[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/ios/devices.dart:169]: My patience is great but my time is short :P

[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/ios/devices.dart:169]: Though - we could accomplish the same thing by moving the build logic out of the devices and always running as if they prebuilt.

[the remark this answers is not in the document]
Comment (jmagman) [reply packages/flutter_tools/lib/src/ios/devices.dart:169]: That's a good idea.

[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/ios/devices.dart:169]: I'm going to look into doing that instead for now. This will be here If we come back to it

**Pull request closed without merging**